plugins/Login/Emails/PasswordResetEmail.php

```diff
@@ -9,85 +9,68 @@
 
 namespace Piwik\Plugins\Login\Emails;
 
-use Piwik\Common;
 use Piwik\Config;
 use Piwik\Mail;
 use Piwik\Piwik;
+use Piwik\View;
 
 class PasswordResetEmail extends Mail
 {
     /**
      * @var string
      */
-    private $login;
+    protected $login;
 
     /**
      * @var string
      */
-    private $ip;
+    protected $ip;
 
     /**
      * @var string
      */
-    private $resetUrl;
+    protected $cancelUrl;
 
-    public function __construct($login, $ip, $resetUrl)
+    /**
+     * @var string
+     */
+    protected $resetUrl;
+
+    public function __construct(string $login, string $ip, string $resetUrl, string $cancelUrl)
     {
         parent::__construct();
 
         $this->login = $login;
         $this->ip = $ip;
         $this->resetUrl = $resetUrl;
+        $this->cancelUrl = $cancelUrl;
 
         $this->setUpEmail();
     }
 
-    private function setUpEmail()
+    private function setUpEmail(): void
     {
         $replytoEmailName = Config::getInstance()->General['login_password_recovery_replyto_email_name'];
         $replytoEmailAddress = Config::getInstance()->General['login_password_recovery_replyto_email_address'];
 
         $this->setSubject($this->getDefaultSubject());
         $this->addReplyTo($replytoEmailAddress, $replytoEmailName);
-        $this->setWrappedHtmlBody($this->getHTMLBody());
-        $this->setBodyText($this->getDefaultBodyText());
+        $this->setWrappedHtmlBody($this->getDefaultBodyView());
```

Comment:
We should still set a `bodyText` for the email. Otherwise it will be automatically set to the HtmlBody with all HTML tags removed. Currently that ends up in a nice text, but the link urls are completely missing. So someone using a text only email client would not directly be able to see the links.